starlite/testing/test_client/client.py

```diff
@@ -598,7 +645,107 @@ def test_something(self, test_client: TestClient) -> None:
+        return self.get_session_data()
+
+    @staticmethod
+    def _create_session_cookies(backend: CookieBackend, data: Dict[str, Any]) -> Dict[str, str]:
+        encoded_data = backend.dump_data(data=data)
+        return {cookie.key: cast("str", cookie.value) for cookie in backend._create_session_cookies(encoded_data, {})}
+
+    async def _set_session_data_async(self, data: Dict[str, Any]) -> None:
+        # TODO: Expose this in the async client
+
+        if isinstance(self.session_backend, ServerSideBackend):
+            serialized_data = self.session_backend.serlialize_data(data)
+            session_id = self.cookies.setdefault(
+                self.session_backend.config.key, self.session_backend.generate_session_id()
+            )
+            await self.session_backend.set(session_id, serialized_data)
+        elif isinstance(self.session_backend, CookieBackend):
+            for key, value in self._create_session_cookies(self.session_backend, data).items():
+                self.cookies.set(key, value)
+        else:
+            raise_for_unsupported_session_backend(self.session_backend)
+
+    async def _get_session_data_async(self) -> Dict[str, Any]:
+        # TODO: Expose this in the async client
+
+        if isinstance(self.session_backend, ServerSideBackend):
+            session_id = self.cookies.get(self.session_backend.config.key)
+            if session_id:
+                data = await self.session_backend.get(session_id)
+                return self.session_backend.deserialize_data(data)
+        elif isinstance(self.session_backend, CookieBackend):
+            raw_data = [
+                self.cookies[key].encode("utf-8") for key in self.cookies if self.session_backend.config.key in key
+            ]
+            if raw_data:
+                return self.session_backend.load_data(data=raw_data)
+        else:
+            raise_for_unsupported_session_backend(self.session_backend)
+        return {}
+
+    def set_session_data(self, data: Dict[str, Any]) -> None:
+        """Set session data.
+
+        Args:
+            data: Session data
+
+        Returns:
+            None
+
+        Examples:
+            ```python
+            from starlite import Starlite, get
+            from starlite.middleware.session.memory_backend import MemoryBackendConfig
+
+            session_config = MemoryBackendConfig()
+
+
+            @get(path="/test")
+            def get_session_data(request: Request) -> Dict[str, Any]:
+                return request.session
+
+
+            app = Starlite(
+                route_handlers=[get_session_data], middleware=[session_config.middleware]
+            )
+
+            with TestClient(app=app, session_config=session_config) as client:
+                client.set_session_data({"foo": "bar"})
+                assert client.get("/test").json() == {"foo": "bar"}
+            ```
+        """
+        anyio.run(self._set_session_data_async, data)
```

Comment:
please destructure run as `anyio_run`, also - why use this rather than simply await the asyn function here?

Reply:
Can you elaborate on the `await`ing?

Reply:
why not simply do:

```python
await self._set_session_data_async(data)
```

instead?

Reply:
Because this is not an asynchronous function? 
`_set_session_data_async` is async, and we want to expose a synchronous interface. That's why I use anyio to run it

Reply:
Happens :D